### PR TITLE
[Integration][AWS] throw error if no permissions on region

### PR DESCRIPTION
**Pull request by shalev007**
# Description

What - We have implemented blacklist error handling for AWS errors in our system. This ensures that certain AWS error types are specifically caught and handled differently from other errors.

Why - Each AWS resource handles "not found" errors differently, making it challenging to distinguish between a resource not being found and an actual error in the system. This inconsistency can lead to misinterpretation of errors and inappropriate error handling. By adding blacklist error handling, we can provide more accurate and consistent error handling across different AWS resources.

How - We have introduced a blacklist mechanism that explicitly catches specific AWS error codes related to "not found" scenarios. These error codes are then handled separately from other types of errors, allowing us to treat them as non-critical issues and proceed accordingly. This ensures that our system can gracefully handle situations where AWS resources are legitimately not found without mistaking them for system errors. This approach improves our system's robustness and reliability in interacting with AWS services.

## Type of change

Please leave one option from the following and delete the rest:

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/aws/main.py [hunk 1]
```diff
@@ -224,7 +225,19 @@ async def webhook(update: ResourceUpdate, response: Response) -> fastapi.Respons
...
+                    )
+                    return fastapi.Response(
+                        status_code=status.HTTP_200_OK,
+                        content=json.dumps({"ok": True}),
+                    )
+                resource = None
+
+            # some aws resources return not found as an empty response
+            if not resource:
                 resource = None
```
Comment (Tankilevitch) [inline integrations/aws/main.py:241]: redundant?

File: integrations/aws/main.py [hunk 2]
```diff
@@ -224,7 +225,19 @@ async def webhook(update: ResourceUpdate, response: Response) -> fastapi.Respons
...
+            except Exception as e:
+                if is_access_denied_exception(e) or is_server_error(e):
+                    logger.error(
+                        f"Skipping resyncing {resource_type} in region {region} due to missing access permissions"
+                    )
+                    return fastapi.Response(
+                        status_code=status.HTTP_200_OK,
+                        content=json.dumps({"ok": True}),
+                    )
+                resource = None
```
Comment (Tankilevitch) [inline integrations/aws/main.py:237]: if access denied we do want to try to delete I think

Comment (shalev007) [reply integrations/aws/main.py:237]: Why is that?

**Review by Tankilevitch — APPROVED** [review]
bump project.toml and version and fix suggestion

**Pull request merged**